### PR TITLE
IAuthRequirementsData 

### LINKED ISSUE #29291: New doc: custom authorization policies with IAuthorizationRequirementData
Create a new doc under [Introduction to authorization in ASP.NET Core](https://learn.microsoft.com/en-us/aspnet/core/security/authorization/introduction?view=aspnetcore-7.0)

See [Improved support for custom authorization policies with IAuthorizationRequirementData](https://devblogs.microsoft.com/dotnet/asp-net-core-updates-in-dotnet-8-preview-4/#improved-support-for-custom-authorization-policies-with-iauthorizationrequirementdata)

Title: Using `IAuthorizationRequirementData` with custom authorization policies

---
#### Document Details

⚠ *Do not edit this section. It is required for learn.microsoft.com ➟ GitHub issue linking.*

* ID: 90aa2423-5082-413f-1009-86b353c27cf6
* Version Independent ID: 84241d82-3261-1225-d1f9-a94d38f6155b
* Content: [Introduction to authorization in ASP.NET Core](https://learn.microsoft.com/en-us/aspnet/core/security/authorization/introduction?view=aspnetcore-7.0)
* Content Source: [aspnetcore/security/authorization/introduction.md](https://github.com/dotnet/AspNetCore.Docs/blob/main/aspnetcore/security/authorization/introduction.md)
* Product: **aspnet-core**
* Technology: **aspnetcore-security**
* GitHub Login: @Rick-Anderson
* Microsoft Alias: **riande**

---
[Associated WorkItem - 90670](https://dev.azure.com/msft-skilling/Content/_workitems/edit/90670)

### PR DESCRIPTION
**Pull request by Rick-Anderson**
Fixes #29291

---

#### Internal previews

| 📄 File | 🔗 Preview link |
|:--|:--|
| [aspnetcore/security/authorization/iard.md](https://github.com/dotnet/AspNetCore.Docs/blob/4c2bb233e9576e08e02bc7297e445127b1606430/aspnetcore/security/authorization/iard.md) | [aspnetcore/security/authorization/iard](https://review.learn.microsoft.com/en-us/aspnet/core/security/authorization/iard?branch=pr-en-us-29374) |

### CODE REVIEW

File: aspnetcore/security/authorization/iard.md
Comment (Rick-Anderson) [inline aspnetcore/security/authorization/iard.md:59]: This will be moved to What's new in 8.0 as soon as this PR merges (to avoid merge conflicts)

**Comment by blowdart** [comment]
I defer to Safia here, as its her changes.

File: aspnetcore/security/authorization/iard.md
Comment (captainsafia) [inline aspnetcore/security/authorization/iard.md:31]:
```suggestion
If a custom policy provider is able to handle all expected policy names, setting the fallback policy with <xref:Microsoft.AspNetCore.Authorization.IAuthorizationPolicyProvider.GetFallbackPolicyAsync> is not required.
```

File: aspnetcore/security/authorization/iard.md
Comment (captainsafia) [inline aspnetcore/security/authorization/iard.md:125]: Can we truncate this code block so that it only shows the method signature? The doc takes a long time to scroll through at the moment.

**Review by captainsafia — COMMENTED** [review]
Is there going to be a reference in the ToC to this doc?

**Comment by Rick-Anderson** [comment]
> Is there going to be a reference in the ToC to this doc?

I'll add it to the ToC and move `## IAuthorizationRequirementData` to the What's new in 8.0 doc as soon as this PR is approved. I did that in two steps to avoid a merge conflict.

**Review by captainsafia — APPROVED** [review]

**Pull request merged**
